tar/chunks/chunks_write.c

```diff
@@ -33,16 +33,20 @@ struct chunks_write_internal {
 	struct chunkstats stats_tape;	/* This archive, w/ multiplicity. */
 	struct chunkstats stats_new;	/* New chunks. */
 	struct chunkstats stats_tapee;	/* Extra data in this archive. */
+	void ** siginfo_cookie;		/* Cookie for siginfo. */
+	void (* siginfo_newbytes) (void **, uint64_t);	/* Set new bytes. */
 };
 
 /**
- * chunks_write_start(cachepath, S, maxchunksize):
+ * chunks_write_start(cachepath, S, maxchunksize, siginfo, siginfo_newbytes):
  * Start a write transaction using the cache directory ${cachepath} and the
  * storage layer cookie ${S} which will involve chunks of maximum size
- * ${maxchunksize}.
+ * ${maxchunksize}.  If ${siginfo_newbytes} is not NULL, call it with
+ * ${siginfo_cookie} and the number of new bytes.
```

Comment:
That last sentence feels like it ended too soon.  "... after writing each chunk"?